### PR TITLE
lambda.part2.exercise2 (Lukovkin)

### PR DESCRIPTION
**Pull request by AlexLuk**

### CODE REVIEW

File: src/test/java/lambda/part2/exercise/Exercise2.java [hunk 1]
```diff
@@ -25,25 +25,25 @@ public void personHasNotEmptyLastNameAndFirstName() {
...
-        throw new UnsupportedOperationException();
+        return p -> !predicate.test(p);
     }
 
     // TODO метод (Person -> boolean, Person -> boolean) -> (Person -> boolean)
     // TODO - возвращает новый предикат, объединяющий исходные с помощью операции "AND"
     // TODO - при реализации использовать логический оператор &&
     private Predicate<Person> andUsingLogicalOperator(Predicate<Person> left, Predicate<Person> right) {
-        throw new UnsupportedOperationException();
+        return left.and(right);
```
Comment (loryrosh) [inline src/test/java/lambda/part2/exercise/Exercise2.java]: should be && operator

File: src/test/java/lambda/part2/exercise/Exercise2.java [hunk 2]
```diff
@@ -54,25 +54,25 @@ public void personHasNotEmptyLastNameAndFirstNameUsingLogicalOperators() {
...
+        Predicate<Person> personHasEmptyFirstName = person -> person.getFirstName().isEmpty();
+        Predicate<Person> personHasEmptyLastName = person -> person.getLastName().isEmpty();
 
-        Predicate<Person> personHasNotEmptyFirstName = null;
-        Predicate<Person> personHasNotEmptyLastName = null;
+        Predicate<Person> personHasNotEmptyFirstName = negate(personHasEmptyFirstName);
+        Predicate<Person> personHasNotEmptyLastName = negate(personHasEmptyLastName);
 
-        Predicate<Person> personHasNotEmptyLastNameAndFirstName = null;
+        Predicate<Person> personHasNotEmptyLastNameAndFirstName = personHasNotEmptyFirstName.and(personHasNotEmptyLastName);
```
Comment (loryrosh) [inline src/test/java/lambda/part2/exercise/Exercise2.java]: should be and(...) method

File: src/test/java/lambda/part2/exercise/Exercise2.java (hunk 1 above)
Comment (AlexLuk) [reply src/test/java/lambda/part2/exercise/Exercise2.java]: Fixed

File: src/test/java/lambda/part2/exercise/Exercise2.java (hunk 2 above)
Comment (AlexLuk) [reply src/test/java/lambda/part2/exercise/Exercise2.java]: Fixed

**Pull request merged**